pkg/client/cli/cliutil/daemon.go

```diff
@@ -0,0 +1,186 @@
+package cliutil
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	// nolint:depguard // Because we won't ever .Wait() for the process and we'd turn off
+	// logging, using dexec would just be extra overhead.
+	"os/exec"
+
+	"github.com/kballard/go-shellquote"
+	"google.golang.org/grpc"
+	empty "google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/datawire/dlib/dexec"
+	"github.com/telepresenceio/telepresence/rpc/v2/daemon"
+	"github.com/telepresenceio/telepresence/v2/pkg/client"
+	"github.com/telepresenceio/telepresence/v2/pkg/filelocation"
+)
+
+var ErrNoDaemon = errors.New("telepresence root daemon is not running")
+
+func launchDaemon(ctx context.Context, dnsIP string) error {
+	fmt.Println("Launching Telepresence Daemon", client.DisplayVersion())
+
+	// Ensure that the logfile is present before the daemon starts so that it isn't created with
+	// root permissions.
+	logDir, err := filelocation.AppUserLogDir(ctx)
+	if err != nil {
+		return err
+	}
+	logFile := filepath.Join(logDir, "daemon.log")
+	if _, err := os.Stat(logFile); err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+		if err = os.MkdirAll(logDir, 0700); err != nil {
+			return err
+		}
+		fh, err := os.OpenFile(logFile, os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			return err
+		}
+		_ = fh.Close()
+	}
+
+	configDir, err := filelocation.AppUserConfigDir(ctx)
+	if err != nil {
+		return err
+	}
+
+	args := []string{client.GetExe(), "daemon-foreground", logDir, configDir, dnsIP}
+	if os.Geteuid() != 0 {
+		// If we're going to be prompting for the `sudo` password, we want to first provide
+		// the user with some info about exactly what we're prompting for.  We don't want to
+		// use `sudo`'s `--prompt` flag for this because (1) we don't want it to be
+		// re-displayed if they typo their password, and (2) it might be ignored anyway
+		// depending on `passprompt_override` in `/etc/sudoers`.  So we'll do a pre-flight
+		// `sudo --non-interactive true` to decide whether to display it.
+		needPwCmd := dexec.CommandContext(ctx, "sudo", "--non-interactive", "--true")
```

Comment:
```suggestion
		needPwCmd := dexec.CommandContext(ctx, "sudo", "--non-interactive", "true")
```

Reply:
Oops, I had meant to change that from `true` to `--validate`, but I only made half the change!